ipaserver/install/ipa_restore.py

```diff
@@ -714,12 +717,16 @@ def remove_old_files(self):
                 if e.errno != 2:  # 2: dir does not exist
                     logger.warning("Could not remove directory: %s (%s)", d, e)
 
-        for f in self.FILES_TO_BE_REMOVED:
+    def clear_old_files(self):
+        """
+        Replace exist files that cannot be removed with zero-length files
+        before backup
+        """
+        for f in self.FILES_TO_BE_CLEARED:
             try:
-                os.remove(f)
-            except OSError as e:
-                if e.errno != 2:  # 2: file does not exist
-                    logger.warning("Could not remove file: %s (%s)", f, e)
+                open(f, 'w').close()
+            except IOError as e:
```

Comment:
``os.remove`` is an alias for ``os.unlink``. The unlink function will never raise IOError.

I think the logic is wrong here, at least for the special case of ``nss.conf``. It should be cleared no matter what its previous state was. The function only clears it when it already exists but doesn't clear it when it is missing. Shouldn't it be created as empty file on both update and restore?

Reply:
> `os.remove` is an alias for `os.unlink`. The unlink function will never raise IOError.

`os.remove` with try-except is a part of original code, it isn't mine. 

I added `OSError` for logging situations when it is not possible to clear file (open it for writing), e.g. no permissions for writing. Actually python3 has `PermissionError` exception (based on `OSError`) for that, but because we are supprting python2 at 4.7 I have used `OSError` 

> Shouldn't it be created as empty file on both update and restore?
Yes, I think you are right. Thanks for an advice.